### PR TITLE
fix: markdown styling improvements, list item newline fix

### PR DESCRIPTION
**Pull request by ghgoodreau**
## Motivation

#60 - markdown styling improvements and list element display bug

## Solution

TBD, utilizing chakra components for each element seems to work. still troubleshooting the newline list bug.

## Checklist

- [x] Tested in Chrome
- [x] Tested in Safari

---

## PR-Codex overview
This PR adds Chakra UI styling and formatting to Markdown components in the app. Notable changes include:

### Detailed summary
- Added Chakra UI `List`, `ListItem`, and `CopyIcon` components
- Adjusted `Markdown` component to include new `List` and `ListItem` components for bullet points and numbered lists
- Adjusted `Markdown` component to include `Box` and `blockquote` components for block quotes
- Adjusted `Markdown` component to filter out empty strings in child elements
- Adjusted `Markdown` component to add left border to code blocks

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by ghgoodreau** [comment]
@transmissions11 try now and see if it looks better on your end. lists and white spaces don't seem to work well. wrapping in a flex component fixes the visual bug but doesn't really display the list whitespace, seems like an alright tradeoff

**Comment by ghgoodreau** [comment]
For example,

```markdown
**Flux** is really _cool_

- test

- test

1. hi

2. hi
```

Now:
Screenshot 2023-04-06 at 4 07 06 PM

Before: 
Screenshot 2023-04-06 at 4 07 18 PM

**Comment by ghgoodreau** [comment]
lol nevermind this causes some issues with markdown elements inside lists

- like *this* element

🙄 need to keep looking into it.

**Comment by ghgoodreau** [comment]
@transmissions11 this should fix the styling issues, had to do some more processing to fix the newline markdown bug but it works

File: src/components/utils/Markdown.tsx [hunk 1]
```diff
@@ -64,12 +73,55 @@ const CopyCodeButton = ({ code }: { code: ReactNode[] }) => {
...
+              <ListItem as="li" mb="0px" ml="20px">
+                {filteredChildren}
+              </ListItem>
+            );
+          },
+          blockquote({ children }) {
+            return (
+              <Box
+                borderLeft="10px solid #eee"
+                backgroundColor="#f5f5f5"
```
Comment (transmissions11) [inline src/components/utils/Markdown.tsx:116]:
```suggestion
                backgroundColor="white"
```

File: src/components/utils/Markdown.tsx [hunk 2]
```diff
@@ -64,12 +73,55 @@ const CopyCodeButton = ({ code }: { code: ReactNode[] }) => {
...
+            return (
+              <ListItem as="li" mb="0px" ml="20px">
+                {filteredChildren}
+              </ListItem>
+            );
+          },
+          blockquote({ children }) {
+            return (
+              <Box
+                borderLeft="10px solid #eee"
```
Comment (transmissions11) [inline src/components/utils/Markdown.tsx:115]:
```suggestion
                borderLeft="5px solid #F5F5F5"
```

File: src/components/utils/Markdown.tsx [hunk 3]
```diff
@@ -64,12 +73,55 @@ const CopyCodeButton = ({ code }: { code: ReactNode[] }) => {
...
+            return (
+              <ListItem as="li" mb="0px" ml="20px">
+                {filteredChildren}
+              </ListItem>
+            );
+          },
+          blockquote({ children }) {
+            return (
+              <Box
+                borderLeft="5px solid #F5F5F5"
```
Comment (transmissions11) [inline src/components/utils/Markdown.tsx:115]:
```suggestion
                borderLeft="8px solid #F5F5F5"
```

**Comment by transmissions11** [comment]
sorry will merge this asap tmrw just dealing with school nonsense and need to sit down or 10 minutes and nitpick a couple things hehe

great work

**Comment by ghgoodreau** [comment]
esketit

**Comment by transmissions11** [comment]
lgtm! sorry for the delay

**Pull request merged**